modules/images/webp-uploads/rest-api.php

```diff
@@ -22,21 +22,27 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
 		return $response;
 	}
 
-	$metadata = wp_get_attachment_metadata( $post->ID );
-	foreach ( $data['media_details']['sizes'] as $size => $details ) {
-		if ( empty( $metadata['sizes'][ $size ]['sources'] ) || ! is_array( $metadata['sizes'][ $size ]['sources'] ) ) {
+	foreach ( $data['media_details']['sizes'] as $size => &$details ) {
```

Comment:
curious why you have the value as a pointer here in this foreach loop and below. Is that required or more performant?

Reply:
Yes, we are changing `source_url` here https://github.com/WordPress/performance/blob/fix/426-enhance-js-replacement-mechanism/modules/images/webp-uploads/rest-api.php#L33

Reply:
that updates `$mime_details`, I don't see `$details` getting modified, it shouldn't need passing by reference?

Reply:
Yes, it updates `$mime_details`. It is part of the `$details` value. Let me share a screenshot of the output with and without pass-by-reference.

With reference to `$details` - https://snipboard.io/CAabcf.jpg

Without reference to `$details` - https://snipboard.io/eGt2I5.jpg

Please correct me if I've made a mistake.

Reply:
@adamsilverstein Both `$details` and `$mime_details` must be passed by reference, otherwise this logic breaks. Value variables in `foreach` loops will otherwise not get updated.